cpp/cmake_modules/DefineOptions.cmake

```diff
@@ -375,6 +375,8 @@ takes precedence over ccache if a storage backend is configured" ON)
 
   define_option(ARROW_PLASMA "Build the plasma object store along with Arrow" OFF)
 
+  define_option(ARROW_WITH_QPL "Enable Intel® Query Processing Library" OFF)
```

Comment:
Could you move this to other `ARROW_WITH_XXX` options like `ARROW_WITH_UCX` and `ARROW_WITH_UTF8PROC`?